rudder-web/src/main/scala/com/normation/rudder/web/services/ComplianceData.scala

```diff
@@ -95,6 +95,163 @@ object RulePolicyMode {
         }
     }
   }
+
+  def directiveModeOnNode(nodeMode : Option[PolicyMode], globalMode : GlobalPolicyMode)(directiveMode : Option[PolicyMode]) = {
+   globalMode.overridable match {
+      case PolicyModeOverrides.Unoverridable =>
+        (globalMode.mode.name, s"""Rudder's global agent policy mode is set to <i><b>${globalMode.mode.name}</b></i> and is not overridable on a per-node or per-directive basis. Please check your Settings or contact your Rudder administrator.""")
+      case PolicyModeOverrides.Always =>
+        val (mode,expl) = matchMode(nodeMode,globalMode,directiveMode)
+        (mode.name,expl)
+    }
+  }
+  def matchMode(nodeMode : Option[PolicyMode], globalMode : GlobalPolicyMode, directiveMode : Option[PolicyMode]) : (PolicyMode,String) = {
```

Comment:
You should add a coment there, saying that it cannot be called with a globalmode not overridable (and probably enforce it in the code)